[Ong Gin Kai Leonard] iP

Frozennfishh opened the pull request:
# Ekud! ฅ^>⩊<^ ฅ
> “I have studied many philosophers and many cats. The wisdom of cats is infinitely superior.” – Hippolyte Taine

The new and improved version of Duke, Ekud is:
- Text-based
- Easy to learn
- ~~Fast~~ **_ULTRA_** FAST to use <br/>
Steps to download:
1. Download **ekud.jar** from [here](https://github.com/Frozennfishh/ip/releases/tag/v0.1)
2. Move it to a folder (e.g. Ekud)
3. Open the terminal and navigate to the folder created
4. Run the command `java -jar "ekud.jar"`
5. Add your tasks.
6. Let it manage your tasks for you 🐈 
And it is **FREE**! <br/>
Features:
- [x] Managing Tasks
- [x] Managing Deadlines, To-dos and Events 
- [x] Searches Tasks in the list by date and keywords
- [x] Automatically saved and easily exportable 
- [ ] Sorting by deadline and/or name (Coming Soon!)

> [!NOTE]
> If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: <br/>
>```ruby
>public class Main {
>    public static void main(String[] args) throws FileNotFoundException {
>        new Ekud("data/list.txt").run();
>    }
>}
>```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Frozennfishh committed:
Add Ekud.java

Frozennfishh committed:
Added intro(), goodbye() methods

Frozennfishh committed:
Added echo

Frozennfishh committed:
Added list function

Frozennfishh committed:
Added mark, unmark function

Frozennfishh committed:
Added todo, event, deadline functions

Frozennfishh committed:
Text Ui Testing

Frozennfishh committed:
Added delete function

Frozennfishh committed:
Added loadFileContent method

Frozennfishh committed:
Added file creation method

Frozennfishh committed:
Added list initalisation

Frozennfishh committed:
Removing data file

Frozennfishh committed:
Added save function

Frozennfishh committed:
Redoing branching

Frozennfishh committed:
Adding load file and save function

Frozennfishh committed:
Added Placeholder.txt

Frozennfishh committed:
Merge branch 'master' into branch-Level-7

Frozennfishh committed:
Merge branch 'master' of https://github.com/Frozennfishh/ip

Frozennfishh committed:
Merge branch 'master' into branch-Level-7

# Conflicts:
#	src/main/java/Ekud.java

Frozennfishh committed:
Remove data file

Frozennfishh committed:
Edited Placeholder.txt

Frozennfishh committed:
Added parseDateTime for Deadline class

Frozennfishh committed:
Added DateTimeParser class

Frozennfishh committed:
Added DateTimeParser in Event Class and added Clear method

Frozennfishh committed:
Separate all classes

Added due function

Frozennfishh committed:
Create Storage Class

Frozennfishh committed:
Merge branch 'branch-A-MoreOOP'

Frozennfishh committed:
Add Command, TaskList, Ui classes

Frozennfishh committed:
Add Parser class

Frozennfishh committed:
Add due command

Frozennfishh committed:
Merge branch 'branch-A-MoreOOP'

Frozennfishh committed:
Delete FileClass file

Frozennfishh committed:
Add ekud package

Frozennfishh committed:
Add memory, parser, tasks, ui packages

Frozennfishh committed:
Merge branch 'branch-A-Packages'

Frozennfishh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Frozennfishh committed:
Adjusted ekud.memory.Storage, ekud.ui.Command, ekud.ui.Ui imports

Frozennfishh committed:
Add test files

Frozennfishh committed:
Edit IndexPairTest

Frozennfishh committed:
Merge branch 'branch-A-JUnit'

Frozennfishh committed:
Merge gradle

Frozennfishh committed:
Added test cases for DateTimeParser, TimeParser

Frozennfishh committed:
Merge branch 'branch-A-JUnit'

Frozennfishh committed:
Package App as JAR file

Frozennfishh committed:
Merge branch 'branch-A-Jar'

Frozennfishh committed:
Add JavaDocs for Ekud, IndexTaskPair, Storage, TaskList

Frozennfishh committed:
Add JavaDocs for Parser, DateTimeParser

Frozennfishh committed:
Add JavaDocs for Deadline, Event, Task, Todo

Frozennfishh committed:
Add JavaDocs to Command, Ui

Frozennfishh committed:
Edited all files to comply with coding standard

Frozennfishh committed:
Added findTask, findTaskPrint and find function

Frozennfishh committed:
Merge branch 'branch-A-JavaDoc'

Frozennfishh committed:
Merge branch 'master' of https://github.com/Frozennfishh/ip

# Conflicts:
#	build.gradle
#	src/main/java/Ekud.java
#	src/main/java/ekud/memory/IndexTaskPair.java
#	src/main/java/ekud/memory/Storage.java
#	src/main/java/ekud/memory/TaskList.java
#	src/main/java/ekud/parser/Parser.java
#	src/main/java/ekud/tasks/Deadline.java
#	src/main/java/ekud/tasks/Event.java
#	src/main/java/ekud/ui/Ui.java

Frozennfishh committed:
Edited Test files to follow Coding Standard

Frozennfishh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ekud/ui/Command.java
#	src/main/java/ekud/ui/Ui.java

Frozennfishh committed:
Fix DateTimeParser parseDate, parseDateTime

Frozennfishh committed:
Edit Ui to allow for readable characters to be used for display

YashvanGH commented on Placeholder.txt:
```diff
@@ -0,0 +1,3 @@
+This is a placeholder for branch-Level-7
+
+This is another placeholder to ensure Level-7 is merged properly
```
Haha! That's smart!

YashvanGH commented on src/main/java/ekud/memory/IndexTaskPair.java:
```diff
@@ -0,0 +1,32 @@
+ * Represents a pair of an index and a task.
+ * This class is used to associate a task with its position in a list.
+ */
+public class IndexTaskPair {
+    private int index;
+    private final Task task;
```
Interesting! Could I ask why you decided to create this class and use it as such?

Frozennfishh replied:
It's to allow for easier tracking of which index is tagged to what task, for now its for printing lists to correspond task to index but in the future can be used for other things!

YashvanGH commented on src/main/java/ekud/memory/TaskList.java:
```diff
@@ -0,0 +1,180 @@
+        return list.isEmpty();
+    }
+
+    /**
+     * Returns the size of the task list.
+     *
+     * @return The number of tasks in the list.
+     */
+    public int size() {
+        return list.size();
+    }
```
I think for user defined methods, we need to follow this rule:
![image](https://github.com/user-attachments/assets/a1e330c8-b136-4c85-b5d9-910514642137)

YashvanGH commented on src/main/java/ekud/parser/DateTimeParser.java:
```diff
@@ -0,0 +1,87 @@
+ * </p>
+ */
+public class DateTimeParser {
+    private static final String[] DATE_TIME_PATTERNS = {
+        "d/M/yyyy HHmm",
+        "dd/MM/yyyy HH:mm",
+        "yyyy-MM-dd HH:mm:ss",
+        "yyyy/MM/dd HH:mm",
+        "dd MMM yyyy HH:mm",
+        "dd MMMM yyyy HH:mm",
+        "EEE, dd MMM yyyy HH:mm",
+        "EEEE, dd MMMM yyyy HH:mm"
+    };
+
+    private static final String[] DATE_PATTERNS = {
+        "d/M/yyyy",
+        "dd/MM/yyyy",
+        "yyyy-MM-dd",
+        "yyyy/MM/dd",
+        "dd MMM yyyy",
+        "dd MMMM yyyy",
+        "EEE, dd MMM yyyy",
+        "EEEE, dd MMMM yyyy"
+    };
```
Really good that you included many ```DATE_TIME_PATTERNS``` and ```DATE_PATTERNS``` to parse a date!

moingshoing replied:
I like how you kept in mind to use both plural form and screaming snake case for the collection of date patterns.

YashvanGH commented on src/main/java/ekud/memory/Storage.java:
```diff
@@ -0,0 +1,110 @@
+        while (s.hasNext()) {
+            String[] line = s.nextLine().split("\\|");
+            System.out.print(i + 1 + ". ");
+            switch (line[0]) {
+            case "T": {
+                list.add(new Todo(line[1], Integer.parseInt(line[2])));
+                break;
+            }
+            case "D": {
+                list.add(new Deadline(line[1], line[2], Integer.parseInt(line[3])));
+                break;
+            }
+            case "E": {
+                list.add(new Event(line[1], line[2], line[3], Integer.parseInt(line[4])));
+                break;
+            }
+            default:
+                break;
+            }
```
Well done for not making the common mistake of indenting your ```case``` in your ```switch``` statement!

YashvanGH commented on src/main/java/ekud/tasks/Event.java:
```diff
@@ -0,0 +1,101 @@
+     * If the start or end time was successfully parsed, it is formatted in
+     * {@code "MMM dd yyyy, h:mm a"} format.
+     * Otherwise, the original input string is displayed.
+     * </p>
+     *
```
Maybe remove one of the empty lines from this JavaDoc for consistency with your other JavaDoc comments?

YashvanGH commented on src/main/java/ekud/tasks/Event.java:
```diff
@@ -0,0 +1,101 @@
+     * The constructor attempts to parse the start and end times using {@code DateTimeParser}.
+     * If a valid format is found, it is stored as a {@code LocalDateTime}.
+     * Otherwise, the original string input is retained.
+     * </p>
+     *
```
Maybe remove one of the empty lines from the JavaDoc for consistency with your other JavaDoc comments?

YashvanGH commented on src/main/java/ekud/tasks/Task.java:
```diff
@@ -0,0 +1,69 @@
+ * It serves as the base class for specific task types such as {@code Todo}, {@code Deadline}, and {@code Event}.
+ * </p>
+ */
+public class Task {
```
Did you consider making your ```Task``` class ```abstract```?

YashvanGH commented on src/main/java/ekud/tasks/Task.java:
```diff
@@ -0,0 +1,69 @@
+ * </p>
+ */
+public class Task {
+    private int done;
```
Would it have been possible to make this a ```boolean``` variable instead?

Frozennfishh replied:
Enumerated the variable, but will change to boolean if necessary down the line!

YashvanGH commented on src/main/java/ekud/ui/Ui.java:
```diff
@@ -0,0 +1,174 @@
+    /**
+     * Displays an introduction message when the program starts.
+     */
+    public void intro() {
+        String logo = """
+
+                EEEEE K  KK U   U DDD   !
+                E     KKK   U   U D  D  !
+                EEEEE K     U   U D   D !
+                E     KKK   UU UU D  D  \s
+                EEEEE K  KK  UUU  DDD   !""";
+        System.out.println("____________________________\n");
+        System.out.println("Hello! I'm\n" + logo);
+        System.out.println("""
+                /|
+                (^.^7 \s
+                |  |\s
+                V__,)/
+                """);
+        System.out.println("What can I do for you?\n");
+        System.out.println("____________________________\n");
+    }
```
Cool Logo! 👍

YashvanGH commented on src/test/java/ekud/EkudTest.java:
```diff
@@ -0,0 +1,4 @@
+package ekud;
+
+public class EkudTest {
+}
```
Good to see that you still have intention to evolve your testing functionality! 👍

YashvanGH commented on src/test/java/ekud/parser/DateTimeParserTest.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    @Test
+    public void parseDateTest1() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("2/12/2019"));
+    }
+
+    @Test
+    public void parseDateTest2() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("02/12/2019"));
+    }
+
+    @Test
+    public void parseDateTest3() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("2019-12-02"));
+    }
+
+    @Test
+    public void parseDateTest4() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("2019/12/02"));
+    }
+
+    @Test
+    public void parseDateTest5() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("02 Dec 2019"));
+    }
+
+    @Test
+    public void parseDateTest6() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("02 December 2019"));
+    }
+
+    @Test
+    public void parseDateTest7() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("Mon, 02 Dec 2019"));
+    }
+
+    @Test
+    public void parseDateTest8() {
+        assertEquals(LocalDateTime.of(2019, 12, 2, 00, 00), DateTimeParser.parseDate("Monday, 02 December 2019"));
```
Really nice and comprehensive tests for ```DateTimeParserTest```!
Perhaps could include better names for test functions?
![image](https://github.com/user-attachments/assets/e921b707-d4fb-45e0-854c-d7d5c7aac7f4)

Frozennfishh replied:
Haha most of these are just placeholder names, thanks for the reminder!

YashvanGH commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,18 @@
+todo eat
+deadline homework /by Friday
+event party /from 7pm /to 9pm
+list
+mark 2
+event
+event jog /from 2pm
+event jog /from 2pm /to 4pm
+deadline
+deadline assignment
+deadline assignment /by Tuesday
+list
+mark 3
+mark 4
+list
+unmark 2
+list
+bye
```
Nice extensive testing for your CLI application! 👍

YashvanGH reviewed (commented):
LGTM! Your code structure looks very clear, intentional and thought out! 
Other than a few nits, I thoroughly enjoyed reviewing your code as I learnt that I could have done some things your way to reduce the amount of time I spent trying to workout the mess I initially coded out. Well done!

ATB! 💯

moingshoing commented on src/main/java/ekud/parser/Parser.java:
```diff
@@ -0,0 +1,78 @@
+     * @param t The {@code TaskList} where the index should exist.
+     * @return {@code true} if the string is not an integer or exceeds the task list size, otherwise {@code false}.
+     */
+    public static boolean indexChecker(String s, TaskList t) {
```
Maybe you can rename this boolean method to sound more like a boolean, such as isValidIndex?

moingshoing commented on src/main/java/ekud/tasks/Event.java:
```diff
@@ -0,0 +1,101 @@
+        DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
+        String s = start != null ? start.format(outputFormat) : this.startString;
+        String e = end != null ? end.format(outputFormat) : this.endString;
+        return "[E][" + (this.getDone() == 1 ? "X" : " ") + "] " + this.getName()
+                + " (from: " + s + " to: " + e + ")";
```
I like how you use a line break here to improve readability.

Frozennfishh committed:
Check for Checkstyle, adjusted some methods with regards to PR
review

Frozennfishh committed:
Add resources/css, /images, /view

Frozennfishh committed:
Add gui package with DialogBox, EkudApplication, MainWindow class
Moved main entry point into Launcher

Frozennfishh committed:
Debug view fxml files

Frozennfishh committed:
All command classes made for each command

Frozennfishh committed:
Adjusted all execute methods to return String object

Frozennfishh committed:
Implement GUI into main code

Frozennfishh committed:
Merge branch 'branch-Level-10'

Frozennfishh committed:
Merge branch 'master' into branch-A-JavaDoc

Frozennfishh committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

Frozennfishh committed:
Checkstyle for new files

Frozennfishh committed:
All JavaDocs in command package

Frozennfishh committed:
Add JavaDocs for all files in exceptions and gui packages

Frozennfishh committed:
Add JavaDocs for all files in memory and parser packages

Frozennfishh committed:
Add JavaDocs for Ui class

Frozennfishh committed:
Merge branch 'branch-A-CodingStandard'

Frozennfishh committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/ekud/command/ClearCommand.java
#	src/main/java/ekud/command/DeadlineCommand.java
#	src/main/java/ekud/command/DeleteCommand.java
#	src/main/java/ekud/command/DueCommand.java
#	src/main/java/ekud/command/EventCommand.java
#	src/main/java/ekud/command/FindCommand.java
#	src/main/java/ekud/command/MarkCommand.java
#	src/main/java/ekud/command/TodoCommand.java
#	src/main/java/ekud/command/UnknownCommand.java
#	src/main/java/ekud/command/UnmarkCommand.java
#	src/main/java/ekud/exceptions/DukeException.java

Frozennfishh committed:
GUI reply box: no change to display color based on command

Little feedback and clarity if command is registered as the correct
type

Adding class type detector in DialogBox.java
and corresponding style types in DialogBox.css files for first 3
types only

Easier means to add different dialog box styles in the future by
editing css files and adding cases to class type detector

Frozennfishh committed:
GUI reply box: change to display color based on some commands only

Only 3 commands have been tagged to change dialog box color

Adding all command class types into detector in DialogBox.java
and corresponding style types in DialogBox.css files

Dialog box color change is done based on currently implemented classes

Frozennfishh committed:
Revert "GUI reply box: change to display color based on some commands only"

This reverts commit 3aaee23e50c61790952ad575de9d7ca6e65baa80.

Frozennfishh committed:
GUI reply box: change to display color only to some commands

Not all classes have been added to be detected by class type detected

Adding add command class type to detector in DialogBox.java
and corresponding style types in DialogBox.css files

Completed color tagging for current commands

Frozennfishh committed:
Background: currently empty

No background implemented

Added background.jpeg with corresponding edit to main.css to
include background

Frozennfishh committed:
Find Method: Currently does not work

Not well implemented into Parser class

Adding parser for "find" and corresponding class

Due method: end date is not split properly

Method does not split input correctly into start and end date

Adjusted array limits to correctly register the end date

Frozennfishh committed:
Assertions does not exist in the application

Not implemented into any classes yet

Adding assertions for required tasklist, ui and storage
for main Ekud.java class

Frozennfishh committed:
Assertions only found in main class

Assertions not well implemented to the rest of the classes

Assertions for task, ui and storage object exist for
each command class added

Frozennfishh committed:
Assertions lacking in parsing methods

Assertions not implemented in processing and ensuring that
required classes are available for parser class to read

Added assertions for tasklist and storage for parser package

Frozennfishh committed:
Code quality improvements not done for all files

Code quality has not been checked for files in program

Improved code quality with better assertions, happy path made more
clear for ClearCommand, DeadlineCommand and DeleteCommand files

Frozennfishh committed:
Code quality improvements not done for most files

Code quality has not been checked for most files in program

Improved code quality with better assertions, happy path made more
clear for DueCommand, EventCommand and FindCommand files

Frozennfishh committed:
Code quality improvements not done for some files

Code quality has not been checked for some files in program

Improved code quality with better assertions, happy path made more
clear for ListCommand, MarkCommand, TodoCommand, UnmarkCommand
files, using OOP methods to get variables from parent class for
DeadlineCommand and DeleteCommand files

Frozennfishh committed:
Code quality improvements not done for some files

Code quality has not been checked for some files in program

*Improved code quality with better assertions, happy path made more
clear for DialogBox, Ekud, MainWindow, IndexTaskPair, Storage,
TaskList files

*Improved code quality and refactoring duplicate code in dueCheck
method

Frozennfishh committed:
Code quality improvements not done for some files

Code quality has not been checked for some files in program

*Improved code quality with better assertions, happy path made more
clear for Parser class, checked for all class code violation and
quality

Frozennfishh committed:
Merge pull request #1 from Frozennfishh/branch-A-Assertions

Added assertions to all relevant files

Frozennfishh committed:
Merge branch 'master' of https://github.com/Frozennfishh/ip

Frozennfishh committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/ekud/command/ClearCommand.java
#	src/main/java/ekud/command/DeleteCommand.java
#	src/main/java/ekud/command/DueCommand.java
#	src/main/java/ekud/command/EventCommand.java
#	src/main/java/ekud/command/FindCommand.java
#	src/main/java/ekud/command/ListCommand.java
#	src/main/java/ekud/command/MarkCommand.java
#	src/main/java/ekud/command/TodoCommand.java
#	src/main/java/ekud/command/UnmarkCommand.java
#	src/main/java/ekud/gui/Ekud.java

Frozennfishh committed:
Merge pull request #2 from Frozennfishh/branch-A-CodeQuality

Adjusted code to adhere to code quality standards

Frozennfishh committed:
FindFreeTimesCommand: absent

Method to find free times not yet implemented

*FindFreeTimesCommand class added
*Included relevant Parser case
*adjusted isInteger check

Frozennfishh committed:
FindFreeTimesCommand: able to parse and split input into minute
and hours

Method to process input has not yet been implemented

*FindFreeTimesCommand class: added execute method with event list
tracking and sorting event list by start date
*TaskList class: added getAllEvents() methods to allow for parsing
for tasks with start and end date and times only

Frozennfishh committed:
FindFreeTimesCommand: able to process and sort events without
displaying proper free time

Method to process input has not yet been implemented

*FindFreeTimesCommand class: added full method to find free time
given amount of time and list of events with start and end time
*Need to implement: regard and disregard tasks on event list
depending on if the event has been done or not

Frozennfishh committed:
Merge branch 'branch-FindFreeTimes'

Frozennfishh committed:
FindFreeTimesCommand: Method not been able detect marked
FindFreeTimesOnCommand: Not implemented to be able to find
free times on a specific date

Method to process input has not yet been implemented

*FindFreeTimesCommand class: Added method to detect marked tasks
and ignore them for free time detection
*FindFreeTimesOnCommand class: Added class constructor

Frozennfishh committed:
Merge branch 'branch-FindFreeTimes'

Frozennfishh committed:
FindFreeTimesOnCommand: Not implemented to be able to find
free times on a specific date

Method to process input has not yet been well implemented

*Added functionality to be able to return time slots with given
date and amount of free time
*Abstracted list accessing methods separately from main execute method

Frozennfishh committed:
Merge branch 'branch-FindFreeTimes'

Frozennfishh committed:
Merge branch 'master' into branch-A-CodingStandard

Frozennfishh committed:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

Frozennfishh committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodeQuality

Frozennfishh committed:
Javadocs missing for some classes

Javadocs have yet to been added to FindFreeTimesCommand and
FindFreeTimesOnCommand classes

*Added necessary Javadocs for all methods in both files
*Minor code coding standard adjustments

Frozennfishh committed:
Merge branch 'branch-A-JavaDoc'

Frozennfishh committed:
Merge branch 'master' into branch-A-CodeQuality

Frozennfishh committed:
Code quality improvements not done for some files

Code quality has not been checked for some files in program

*Improved code quality with better assertions, happy path made more
clear for Parser class, checked for all class code violation and
quality
*Abstracted out repeated methods for clearer code
*Added some missing javadocs

Frozennfishh committed:
Merge branch 'branch-A-CodeQuality'

Frozennfishh committed:
Test Cases are not thorough for most files

Tests are not yet implemented for most files.

* Added test cases for ParserTest for parsing date, minutes and hour
* DeadlineTest added to test deadlines with dates and no dates
* Refined some test cases for DateTimeParserTest

Frozennfishh committed:
Merge branch 'branch-A-MoreTesting'

Frozennfishh committed:
No User Guide given.

User guide not yet implemented as project has been in
work in progress

* User guide added with instructions to Ekud
* Ui.png added for grading purposes

Frozennfishh committed:
Formatting of some lines of README displays wrongly on landing page

Some of the lines in the README are not reflected properly
when reflected onto the landing page

Additions of line breaks for the steps to download has been added

Frozennfishh committed:
Formatting of some lines of README displays wrongly on landing page

Some of the lines in the README are not reflected properly
when reflected onto the landing page

Readjustments for the numbering for the steps
to download has been added

Frozennfishh committed:
Gradle build: Fails some coding standard test cases

Some code fails the gradle build coding standard

* Adjustments made to ParserTest and DeadlineTest to adhere
to coding standards
* Finalising and checking other files for undetected coding
standard violations for final build

Frozennfishh committed:
User Guide: Guide has some formatting issues

User guide is not properly displayed on landing page

Some adjustments to underlining and italicizing made.

Frozennfishh committed:
Fix formatting issues for User Guide.

User guide is not properly displayed on landing page

Some adjustments to underlining and italicizing made.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.

Frozennfishh committed:
Fix formatting issues for User Guide

User guide is not properly displayed on landing page.

Some adjustments to typos and punctuation.
